src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const transformedState = Object.assign({}, state);
+  const transformedArray = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(transformedState, action.extraData);
+        break;
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete transformedState[key];
+        };
+        break;
+      case 'clear':
+        for (const key in transformedState) {
+          delete transformedState[key];
+        }
+        break;
+    }
```

Comment:
It is better to add a default case to handle situations when action type didn't match any of cases.
![image](https://user-images.githubusercontent.com/96409563/211208740-4c255ffc-0702-4a10-92d6-ef27446e48a3.png)